wgengine/magicsock/magicsock_test.go

```diff
@@ -497,22 +520,26 @@ func TestTwoDevicePing(t *testing.T) {
 				t.Logf("cfg1: %v", uapi2)
 			}
 		}()
-		pingSeq(t, 20)
+		pingSeq(t, 20, 0)
 	})
 
-	// Put one real route.
+	dev1.RemoveAllPeers()
+	dev2.RemoveAllPeers()
+
+	// Give one peer a non-DERP endpoint. We expect the other to
+	// accept it via roamAddr.
 	cfgs[0].Peers[0].Endpoints = ep0
 	if ep2 := cfgs[1].Peers[0].Endpoints; len(ep2) != 1 {
 		t.Errorf("unexpected peer endpoints in dev2: %v", ep2)
 	}
-	if err := dev1.Reconfig(&cfgs[0]); err != nil {
+	if err := dev2.Reconfig(&cfgs[1]); err != nil {
 		t.Fatal(err)
 	}
-	if err := dev2.Reconfig(&cfgs[1]); err != nil {
+	if err := dev1.Reconfig(&cfgs[0]); err != nil {
 		t.Fatal(err)
 	}
 	t.Run("one real route is enough thanks to spray", func(t *testing.T) {
-		pingSeq(t, 50)
+		pingSeq(t, 50, 700*time.Millisecond)
```

Comment:
skip this test on testing.Short() if it's slow?

Reply:
Eh, this one test case takes ~2-3s with or without this change. I'd much rather make sure that magicsock tests run as much as possible, because empirically this is break-ful code and I don't want to give people the opportunity to miss bugs.

Reply:
I see 300 ms now.

Can this be t.Parallel at least? (only helps if other things are also parallel)